### PR TITLE
feat: implement REPLACE interceptor

### PR DESCRIPTION
**Pull request by waynexia**
# Which issue does this PR close?

Closes #

# Rationale for this change
 

Query result sometimes contains an unavoidable but also unimportant non-fix string, like the time it executes, or some local directories. This interceptor provides a way to replace them.

# What changes are included in this PR?

- interceptor REPLACE and relative test/example
- bugfix: trim begin white space of concat-ed query string.

# Are there any user-facing changes?

New interceptor REPLACE

# How does this change test

example and unit test

### CODE REVIEW

File: sqlness/examples/interceptor-replace/simple/replace.sql [hunk 1]
```diff
@@ -0,0 +1,8 @@
+-- SQLNESS REPLACE 00
+SELECT 0;
+
+-- SQLNESS REPLACE 00
+SELECT 00;
+
+-- SQLNESS REPLACE 0 1
```
Comment (jiacai2050) [inline sqlness/examples/interceptor-replace/simple/replace.sql:7]: - https://docs.rs/regex/1.8.1/regex/#example-replacement-with-named-capture-groups

How about add an example with capture groups?

File: sqlness/src/interceptor/replace.rs [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+pub struct ReplaceInterceptorFactory;
+
+impl InterceptorFactory for ReplaceInterceptorFactory {
+    fn try_new(&self, interceptor: &str) -> Option<InterceptorRef> {
+        if interceptor.starts_with(PREFIX) {
+            let args = interceptor
+                .trim_start_matches(PREFIX)
+                .trim_start()
+                .trim_end();
+            let mut args = args.splitn(2, ' ');
```
Comment (jiacai2050) [inline sqlness/src/interceptor/replace.rs:59]: Add TODO comments to support pattern with blanks?

File: Makefile [hunk 3]
```diff
@@ -21,11 +21,19 @@ check-license:
 clippy:
 	cd $(DIR); cargo clippy --all-targets --all-features --workspace -- -D warnings
 
+cli-test:
+	cd $(DIR)/sqlness-cli;  cargo run -- -c tests -i 127.0.0.1 -p 3306 -u root -P 1a2b3c -d public
+
+example: basic-example bad-example interceptor-arg-example interceptor-replace-example
```
Comment (jiacai2050) [inline Makefile]: Run this target in CI?

File: sqlness/examples/interceptor_replace.rs
Comment (jiacai2050) [inline sqlness/examples/interceptor_replace.rs:1]: This example seems lots of duplicated codes,  how about reuse interceptor_arg.rs by rename to `interceptor_demo.rs`?

Comment (waynexia) [reply sqlness/examples/interceptor_replace.rs:1]: There are many duplications, but most of them are declarations. Thus I think it's acceptable, as each file is a single standalone example. WDYT?

**Review by jiacai2050 — APPROVED** [review]
LGTM

**Comment by waynexia** [comment]
I plan to bump minor version after other interceptors are implemented

**Comment by jiacai2050** [comment]
👍 

Just don't forget update docs, there are little docs about interceptor.

**Pull request merged**